File: Kotlin/Graphs/Graph.kt
abstract class Graph {
    data class Edge(var from:Int, var to:Int, var cost:Int)

    var edges = ArrayList<Edge>()
    var g:Array<MutableList<Int>>
    var n:Int

    constructor(_n :Int) {
    	n = _n
        g = Array<MutableList<Int>>(n, {_ -> mutableListOf()})
    }

    abstract fun add(from:Int, to:Int, cost:Int = 1)
}

File: CPP/Graphs/Graph.cpp
// Graph
//
template <typename T>
class Graph {
  public:
    struct Edge {
        int from;
        int to;
        T cost;
    };

    vector<Edge> edges;
    vector<vector<int>> g;
    int n;

    Graph(int _n) : n(_n) {
        g.resize(n);
    }

    virtual int add(int from, int to, T cost) = 0;
};